src/invocation/ConnectionAuthenticator.ts

```diff
@@ -53,7 +53,6 @@ export class ConnectionAuthenticator {
                 const authResponse = ClientAuthenticationCodec.decodeResponse(msg);
                 switch (authResponse.status) {
                     case AuthenticationStatus.AUTHENTICATED:
-                        this.connection.setAddress(authResponse.address);
```

Comment:
as far as I understand this line is the fix. Actually, there are deeper problems related to mixed usages, like (localhost <-> 127.0.0.1)
We have analyzed the problem in more detail here. 
https://hazelcast.atlassian.net/wiki/spaces/EN/pages/1199407524/Connection+Strategies+-+Blue+Green+by+way+of+CNAME+Configuration+Design#ConnectionStrategies-Blue/GreenbywayofCNAMEConfigurationDesign-Membersandclientswithdifferentinterfacesforsameaddress
I am preparing an extensive fix https://github.com/hazelcast/hazelcast/pull/14104

But currently, we are not supporting mixed usages. It is users responsibility to configure members and clients consistently. I think instead of making this fix, we need to fix the test for now. 
We will work on mixed scenarios as a team and decide on behavior together. Before that, clients should behave the same.